src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -792,77 +803,123 @@ static int parse_imgpath(const std::string &imgpath)
   return 0;
 }
 
-static int do_list_mapped_devices()
+static int get_mapped_info(int pid)
 {
-  char path[64];
-  int m = 0;
-  int fd[2];
+  int r;
+  std::string path = "/proc/" + stringify(pid) + "/cmdline";
+  std::ifstream ifs;
+  std::string cmdline;
+  std::vector<const char*> args;
 
-  if (socketpair(AF_UNIX, SOCK_STREAM, 0, fd) == -1) {
-    int r = -errno;
-    cerr << "rbd-nbd: socketpair failed: " << cpp_strerror(-r) << std::endl;
-    return r;
+  ifs.open(path.c_str(), std::ifstream::in);
+  assert (ifs.is_open());
+  ifs >> cmdline;
+
+  for (unsigned i = 0; i < cmdline.size(); i++) {
+    const char *arg = &cmdline[i];
+    if (i == 0) {
+      if (strcmp(basename(arg) , "rbd-nbd") != 0) {
+        return -EINVAL;
+      } 
```

Comment:
@liupan1111 Noticed in my editor, there is an extra space at the end of line here and below. Could you please remove.